sched/semaphore/sem_tickwait.c

```diff
@@ -68,12 +68,10 @@
 
 int nxsem_tickwait(FAR sem_t *sem, uint32_t delay)
 {
-  FAR struct tcb_s *rtcb = this_task();
+  FAR struct tcb_s *rtcb;
   irqstate_t flags;
   int ret;
 
-  DEBUGASSERT(sem != NULL && up_interrupt_context() == false);
```

Comment:
why remove DEBUGASSERT from nxsem_tickwait? it's always wrong to call nxsem_tickwait inside the interrupt context.

Reply:
1. tickwait should be safe if timeout delay is 0 in ISR context
2. timeout not equal 0 will handle assert in line 115 nxsem_wait()

Reply:
Compared with irregular heap access, shared template and will make it more deterministic